spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/security/IapAuthenticationAutoConfigurationTests.java

```diff
@@ -88,33 +86,31 @@
 
   @Mock static MetadataProvider mockMetadataProvider;
 
-  @Before
-  public void httpRequestSetup() {
+  @BeforeEach
+  void httpRequestSetup() {
     when(this.mockIapRequest.getHeader("x-goog-iap-jwt-assertion")).thenReturn("very fake jwt");
```

Comment:
By the time we have to enable LENIENT mode on half the tests, it makes sense to move this setup from being global to only being done for each test that needs it. Especially since the setup is only one line.

Reply:
Removed the global set up and tried to put it in the tests that needs it along with removing the lenient mode as well but it is still giving "UnnecessaryStubbing" error.

Reply:
Did you add setup to the tests _with_ lenient or to the tests _without_? 
The setup should be needed in tests that did not have the lenient setup.